app/controllers/application_controller.rb

```diff
@@ -5,6 +5,11 @@ def require_user
   	redirect_to sign_in_path unless current_user
   end
 
+  def require_admin
+    flash[:error] = "You do not have permission to do that." if !current_user.admin?
+    redirect_to home_path unless current_user.admin?
+  end
```

Comment:
You probably wanted something like this:

```ruby
unless current_user.admin?
  flash[:error] = "You do not have permission to do that."
  redirect_to home_path
end
```

Rather than each line having `unless current_user.admin?`